### PR TITLE
feat(whatsapp): upgrade SDK to 2.0.0, add pytest unit + integration tests

### PR DESCRIPTION
**Pull request by Tram-Nguyen87**
## Summary

- Bumps `autohive-integrations-sdk` to `~=2.0.0` and converts all `context.fetch()` callers in `whatsapp.py` to the new `FetchResponse` shape (`response.data`).
- Replaces every error-path `ActionResult(data={"error": ...})` with `ActionError(message=...)` and strips `error` / `result` properties from `config.json` output schemas.
- Replaces the legacy asyncio testbed (`test_whatsapp.py`) with a full pytest suite: **118 unit tests** (mocked fetch via `FetchResponse`) and **9 integration tests** (live Graph API via `aiohttp`) with a `destructive` marker on write tests.
- Refreshes `README.md` to drop v1-era `result: bool` / `error: str` output rows, replace the legacy test runner with the pytest pattern, and clarify that `phone_number_id` is a per-action input (not an auth field).

## Changes

**SDK 2.0.0 upgrade**
- `whatsapp/requirements.txt` bumped to `autohive-integrations-sdk~=2.0.0`
- `whatsapp/config.json` version bumped to `2.0.0`
- All 4 `context.fetch()` call sites in `whatsapp.py` now use `response.data` for the body
- All error returns now use `ActionError(message=...)` (validation, API error envelopes, exception handlers, missing-token paths)
- Output schemas in `config.json` tightened: `error` / `result` fields removed; only documented success fields remain (`message_id`, `status`, `quality_rating`)

**Tests**
- Removed `tests/context.py` and the legacy `tests/test_whatsapp.py`
- Added `tests/conftest.py` for `sys.path` setup
- Added `tests/test_whatsapp_unit.py` — 118 tests covering:
  - Every helper function (`get_whatsapp_creds`, `_extract_api_error`, `validate_phone_number`, `validate_phone_number_id`, `validate_media_url`) with parametrized edge cases
  - Every action: happy path, request URL/method, payload structure, headers, validation errors, API errors, exceptions, missing-token, non-dict responses, empty-list responses, plus media-type-specific behavior (audio doesn't get caption, filename only added for documents, etc.)
  - Mocks use accurate HTTP status codes (`200` for success, `400`/`404` for error envelopes)
  - Platform OAuth auth envelope `{"auth_type": "PlatformOauth2", "credentials": {...}}`
- Added `tests/test_whatsapp_integration.py` — 9 live tests:
  - Read-only: `get_phone_number_health` (status + quality rating, invalid ID, nonexistent ID), plus input-validation paths for `send_message` and `send_media_message` that short-circuit before fetch
  - `@pytest.mark.destructive`: one test each for `send_message`, `send_template_message`, `send_media_message` (sends real messages to `WHATSAPP_RECIPIENT_PHONE` — review before running)
- `real_fetch` uses real `resp.status` + `resp.headers`, passes through `params=`, and falls back to `resp.text()` if the body isn't JSON

**Docs**
- `whatsapp/README.md` — added global "Error handling" note explaining `ActionError`; removed `success` / `error` output rows from every action; corrected provider name (`WhatsApp` → `WhatsApp Business`); rewrote Authentication section to clarify `phone_number_id` is a per-action input rather than an auth field; replaced legacy `python tests/test_whatsapp.py` instructions with the pytest unit + integration pattern (safe command first, destructive command second with warning)
- `.env.example` — added `WHATSAPP_*` section with the 6 variables the integration tests read (`ACCESS_TOKEN`, `PHONE_NUMBER_ID`, `RECIPIENT_PHONE`, `TEMPLATE_NAME`, `TEMPLATE_LANG`, `MEDIA_URL`)

## Validation

Local build pipeline (all green):
- `python ../autohive-integrations-tooling/scripts/validate_integration.py whatsapp` → 0 errors, 0 warnings
- `python ../autohive-integrations-tooling/scripts/check_code.py whatsapp` → all 9 sub-checks pass (syntax, imports, JSON, ruff lint, ruff format, bandit, pip-audit, config-code sync, fetch patterns)
- `python -m pytest whatsapp/ -v` → 118 passed
- `ruff check` + `ruff format --config ../autohive-integrations-tooling/ruff.toml whatsapp` → clean

## Test plan

- [x] `pytest whatsapp/tests/test_whatsapp_unit.py -v` passes 118/118
- [x] `pytest whatsapp/tests/test_whatsapp_integration.py -m "integration and not destructive"` passes with a valid `WHATSAPP_ACCESS_TOKEN` + `WHATSAPP_PHONE_NUMBER_ID`
- [x] `pytest whatsapp/tests/test_whatsapp_integration.py -m "integration and destructive"` passes after setting `WHATSAPP_RECIPIENT_PHONE` (sends real messages — review first)

image

### CODE REVIEW

**Comment by Tram-Nguyen87** [comment]
@Tram-Nguyen87 ran the full live e2e tests locally — 9/9 integration tests pass (read-only `get_phone_number_health` + input validation green; destructive sends covered text message, `hello_world` template, and image media to a test recipient). All 118 unit tests also passing.

File: whatsapp/whatsapp.py [hunk 1]
```diff
@@ -121,67 +105,42 @@ async def execute(self, inputs: Dict[str, Any], context: ExecutionContext):
         language_code = inputs.get("language_code", "en")
```
Comment (TheRealAgentK) [inline whatsapp/whatsapp.py]: `send_template_message` defaults `language_code` to `"en"`, but the live destructive test harness defaults `WHATSAPP_TEMPLATE_LANG` to `"en_US"` for the default `hello_world` template. That means the tested live path and the production omitted-input path can send different payloads. Can we align code/config/README/tests here — either default to `en_US` everywhere, or make `language_code` required so callers must provide the approved locale?

File: whatsapp/README.md [hunk 2]
```diff
@@ -51,9 +49,7 @@ Configure the integration within Autohive using platform authentication for What
   - `language_code` (optional): Template language code (default: "en")
```
Comment (TheRealAgentK) [inline whatsapp/README.md]: Please update this default to match the code/config decision above. Right now the docs say omitted `language_code` means `"en"`, but the live template test path uses `"en_US"`.

File: whatsapp/README.md
Comment (TheRealAgentK) [inline whatsapp/README.md:1]: The README/config descriptions still say the integration can “manage conversations”, but the current action surface only sends messages/templates/media and checks phone-number health. Since this PR is refreshing the public docs, can we trim this wording so the README/catalog matches the actual actions? I’d do the same for the config description.

File: whatsapp/config.json
Comment (TheRealAgentK) [inline whatsapp/config.json:1]: Same default mismatch as the action code: config advertises `"en"` for `send_template_message.language_code`, while the live test default is `"en_US"`. Please keep the schema default in sync with whichever behavior we want to support.

**Review by TheRealAgentK — COMMENTED** [review]
Nice SDK v2 migration overall. I left a few comments on aligning the template language default and trimming docs/catalog wording to the actual action surface.

**Comment by Tram-Nguyen87** [comment]
Addressed all four threads and here is a general summary:
  - **language_code** is now a required input rather than defaulting to anything (rationale in thread 1)
  - README + config descriptions trimmed to match the actual action surface
  - **.env.example** merge conflict with master resolved (kept both WhatsApp + YouTube blocks)

**Review by TheRealAgentK — APPROVED** [review]
All my comments have been addressed, this LGTM now from my point of view. Thx @Tram-Nguyen87
